File: fixtures/scale/dart/medium/lib/util/double_compare.dart
const epsilon = 1e-12;

bool nearlyZero(double value) => value.abs() < epsilon;

bool nearlyEqual(double a, double b) {
  final scale = a.abs() > b.abs() ? a.abs() : b.abs();
  return (a - b).abs() < epsilon * (scale > 1.0 ? scale : 1.0);
}

File: fixtures/scale/kotlin/medium/src/calc/util/DoubleCompare.kt
package calc.util

import kotlin.math.abs
import kotlin.math.max

const val EPSILON = 1e-12

fun nearlyZero(value: Double): Boolean = abs(value) < EPSILON

fun nearlyEqual(a: Double, b: Double): Boolean =
    abs(a - b) < EPSILON * max(1.0, max(abs(a), abs(b)))
